tests/framework/artifacts.py

```diff
@@ -118,8 +134,117 @@ def copy(self):
                         local_folder=local_folder)
 
 
+class SnapshotArtifact:
+    """Manages snapshot S3 artifact objects."""
+
+    def __init__(self,
+                 bucket,
+                 key,
+                 artifact_type=ArtifactType.SNAPSHOT):
+        """Initialize bucket, key and type."""
+        self._bucket = bucket
+        self._type = artifact_type
+        self._key = key
+
+        self._mem = Artifact(self._bucket, "{}vm.mem".format(key),
+                             artifact_type=ArtifactType.MISC)
+        self._vmstate = Artifact(self._bucket, "{}vm.vmstate".format(key),
+                                 artifact_type=ArtifactType.MISC)
+        self._ssh_key = Artifact(self._bucket, "{}ssh_key".format(key),
+                                 artifact_type=ArtifactType.SSH_KEY)
+        self._disks = []
+
+        disk_prefix = "{}disk".format(key)
+        snaphot_disks = self._bucket.objects.filter(Prefix=disk_prefix)
+
+        for disk in snaphot_disks:
+            artifact = Artifact(self._bucket, disk.key,
+                                artifact_type=ArtifactType.DISK)
+            # artifact.download()
+            self._disks.append(artifact)
+
+        # Get the name of the snapshot folder.
+        snapshot_name = self.name
+        self._local_folder = os.path.join(ARTIFACTS_LOCAL_ROOT,
+                                          self.type.strip('/'),
```

Comment:
If `self.type` returns the default `ArtifactType.SNAPSHOT` value, the return value does not have `strip()` as an attribute.